### ISSUE TITLE
Duplicate Resource Declaration for Java Package

### ISSUE THREAD

**Issue by yachub** [puppetlabs/puppetlabs-patching_as_code#32]
Thanks for providing the module, it's working really well so far except for one small issue that I'm having trouble pinpointing.

## Describe the Bug
The patching_as_code module is unable to `ensure => latest` on the package `openjdk-7-jre-headless` due to a duplicate resource declaration.

## Expected Behavior
The patching_as_code module should `ensure => latest` and be able to update the java package.

## Steps to Reproduce
1. Use the [puppetlabs-java](https://forge.puppet.com/modules/puppetlabs/java/6.5.0) module to manage the java installation on a node, such as:
```puppet
class { 'java':
  distribution => 'jre',
}
```
2. Ensure the a newer java package is available than what is currently installed.
3. Classify the node in a `patching_as_code::patch_group`.
4. When the node reaches the time frame specified in `patching_as_code::patch_schedule`, every puppet run results in the following error:
```
Failed to apply catalog: Cannot alias Package[openjdk-7-jre-headless]
to [nil, "openjdk-7-jre-headless", :apt]; resource ["Package", nil, "openjdk-7-jre-headless", :apt]
already declared (file: /opt/puppetlabs/server/data/puppetserver/filesync/client/versioned-dirs/puppet-
code/production_18b2b688aed30a89a6f194510d72258fa04fae5b/modules/java/manifests/init.pp, line: 123)
```

## Environment
 - puppetlabs/patching_as_code Module Version: 0.7.1
 - puppetlabs/java Module Version: 6.5.0
 - Platform: Debian 8
 - Puppet Version: 6.19

## Additional Context
Oddly it seems like this issue should not be occurring since the java module declares the package resource as [present](https://github.com/puppetlabs/puppetlabs-java/blob/58edff7fd0ff556c78fceeaf8bf5ee8ba60d000c/manifests/init.pp#L123) and the patching_as_code module should be changing it to [latest](https://github.com/puppetlabs/puppetlabs-patching_as_code/blob/1b225e5164c84ab9f08e3255ea9f082e6f5b9826/lib/puppet/type/patch_package.rb#L51).

**Comment by kreeuwijk** [comment]
@yachub thanks for the bug report, I'll take a look. It seems that the way the package is declared:
```
package { 'java':
    ensure          => $version,
    install_options => $package_options,
    name            => $use_java_package_name,
}
```
is causing it to evade the existing package detection somehow.

**Comment by kreeuwijk** [comment]
@yachub fixed in #33 which is now released as v0.7.3 on the Puppet Forge
